Assets/Scripts/Gameplay/Action/ActionPlayers/ClientActionPlayer.cs

```diff
@@ -30,39 +30,39 @@ public void OnUpdate()
             for (int i = m_PlayingActions.Count - 1; i >= 0; --i)
             {
                 var action = m_PlayingActions[i];
-                bool keepGoing = action.Anticipated || action.OnUpdate(); // only call OnUpdate() on actions that are past anticipation
-                bool expirable = action.Description.DurationSeconds > 0f; //non-positive value is a sentinel indicating the duration is indefinite.
-                bool timeExpired = expirable && action.TimeRunning >= action.Description.DurationSeconds;
-                bool timedOut = action.Anticipated && action.TimeRunning >= k_AnticipationTimeoutSeconds;
+                bool keepGoing = action.AnticipatedClient || action.OnUpdateClient(Parent); // only call OnUpdate() on actions that are past anticipation
+                bool expirable = action.Config.DurationSeconds > 0f; //non-positive value is a sentinel indicating the duration is indefinite.
+                bool timeExpired = expirable && action.TimeRunning >= action.Config.DurationSeconds;
+                bool timedOut = action.AnticipatedClient && action.TimeRunning >= k_AnticipationTimeoutSeconds;
                 if (!keepGoing || timeExpired || timedOut)
                 {
-                    if (timedOut) { action.Cancel(); } //an anticipated action that timed out shouldn't get its End called. It is canceled instead.
-                    else { action.End(); }
+                    if (timedOut) { action.CancelClient(Parent); } //an anticipated action that timed out shouldn't get its End called. It is canceled instead.
```

Comment:
naming "parent" feels misleading. Are we talking about transform parent? netcode parent? action player?
(it's actually none of the above, it's the visualization...)